staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go

```diff
@@ -129,9 +129,17 @@ func (t *azureFileCSITranslator) TranslateInTreePVToCSI(pv *v1.PersistentVolume)
 			resourceGroup = v
 		}
 	}
+
+	// Secret is required when mounting a volume but pod presence cannot be assumed - we should not try to read pod now.
 	namespace := defaultSecretNamespace
```

Comment:
Can you remove this default line? It doesn't look like there is a path for intree PV to use the "default" namespace.

Reply:
Correct, intree does not use "default". So to keep it consistent we should return an error here as well if the namespace can not be found. See if the latest version of the patch would be ok.

Reply:
what if `pv.Spec.ClaimRef.Namespace` is empty? I think it's better not return error.

Reply:
In CSI driver emtpy equals `default` as you mentioned previously, and I think Michelle wants to prevent using `default` as this might point to a secret in different namespace. So I think the only option here is to return error because we can't use any reasonable default. 

Also, how possible is it to have `pv.Spec.ClaimRef.Namespace` empty? It should not be possible to use/mount a PV without PVC correct? So ideally the error should never be returned, but if it ever happens at least we know it's not possible to continue.

And if we don't want to return an error, what should the namespace be set to? We can't leave it empty due to CSI driver interpreting it as `default`, we can't set it to nil because that might break volumeID now (namespace was added recently #107575)), and we can't just leave it out because pv would never mount (without a secret).

Reply:
In theory, for all PV operations `PV.claimRef` must be set (i.e. a PV must be bound).

Attach/detach/mount/unmount work with *PVCs*, i.e. the PV they get must be bound.
Provisioner creates PVs with `claimRef` populated.
Deletetion works when a PV has a `claimRef` set and the referenced PVC does not exist.

PVs without a `claimRef` must have been created manually and are not bound. Does Kubernetes do anything witch such volumes that needs translation? Did I miss any exotic error case?